api/api/src/test/scala/uk/ac/wellcome/platform/api/services/CollectionServiceTest.scala

```diff
@@ -0,0 +1,179 @@
+
+  it("Retrieves a tree with the given path and all ancestors expanded") {
+    withLocalWorksIndex { index =>
+      storeWorks(index)
+      val service = collectionService(index)
+      whenReady(service.retrieveTree(List("a/b"))) { result =>
+        result shouldBe Right(
+          CollectionTree(
+            path = "a",
+            work = workA,
+            children = List(
+              CollectionTree(
+                path = "a/b",
+                work = workB,
+                children = List(
+                  CollectionTree(path = "a/b/c", work = workC),
+                  CollectionTree(path = "a/b/d", work = workD),
+                )
+              ),
+              CollectionTree(path = "a/e", work = workE),
+            )
+          )
+        )
+      }
+    }
+  }
+
+  it("Retrieves a tree with multiple paths and their ancestors expanded") {
+    withLocalWorksIndex { index =>
+      storeWorks(index)
+      val service = collectionService(index)
+      whenReady(service.retrieveTree(List("a/b", "a/e/f"))) { result =>
+        result shouldBe Right(
+          CollectionTree(
+            path = "a",
+            work = workA,
+            children = List(
+              CollectionTree(
+                path = "a/b",
+                work = workB,
+                children = List(
+                  CollectionTree(path = "a/b/c", work = workC),
+                  CollectionTree(path = "a/b/d", work = workD),
+                )
+              ),
+              CollectionTree(
+                path = "a/e",
+                work = workE,
+                children = List(
+                  CollectionTree(
+                    path = "a/e/f",
+                    work = workF,
+                    children = List(
+                      CollectionTree(path = "a/e/f/g", work = workG)
+                    )
+                  ),
+                )
+              ),
+            )
+          )
+        )
+      }
+    }
+  }
+
+  it("Only expands by a single depth beyond the given path") {
+    withLocalWorksIndex { index =>
+      storeWorks(index)
+      val service = collectionService(index)
+      whenReady(service.retrieveTree(List("a"))) { result =>
+        result shouldBe Right(
+          CollectionTree(
+            path = "a",
+            work = workA,
+            children = List(
+              CollectionTree(path = "a/b", work = workB),
+              CollectionTree(path = "a/e", work = workE),
+            )
+          )
+        )
+      }
+    }
+  }
+
+  it("Fails creating a tree when incomplete data") {
+    withLocalWorksIndex { index =>
+      storeWorks(index)
+      val service = collectionService(index)
+      whenReady(service.retrieveTree(List("x/oops"))) { result =>
+        result shouldBe a[Left[_, _]]
+        result.left.get.getMessage shouldBe "Not all works in collection are connected to root 'x': x/oops/z"
+      }
+    }
+  }
```

Comment:
To double-check I’ve understood correctly: this is because there’s no value `x/oops` in here?

Reply:
Yes that's right